docs/buildstrategies.md

```diff
@@ -155,4 +159,313 @@ kubectl apply -f samples/buildstrategy/source-to-image/buildstrategy_source-to-i
+
+### Examples of Tekton resources management
+
+For a more concrete example, lets take a look on the following scenarios:
+
+---
+
+**Scenario 1.**  Namespace without `LimitRange`, both steps with the same resource values.
+
+If we will apply the following resources:
+
+- [buildahBuild](../samples/build/buildah/build_buildah_cr.yaml)
+- [buildahBuildRun](../samples/buildrun/buildah/buildrun_buildah_cr.yaml)
+- [buildahClusterBuildStrategy](../samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml)
+
+We will see some differences between the `TaskRun` definition and the `pod` definition.
+
+For the `TaskRun`, as expected we can see the resources on each `step`, as we previously define on our [strategy](../samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml).
+
+```sh
+$ kubectl -n test-build get tr buildah-golang-buildrun-9gmcx-pod-lhzbc -o json | jq '.spec.taskSpec.steps[] | select(.name == "step-buildah-bud" ) | .resources'
+{
+  "limits": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "requests": {
+    "cpu": "250m",
+    "memory": "65Mi"
+  }
+}
+
+$ kubectl -n test-build get tr buildah-golang-buildrun-9gmcx-pod-lhzbc -o json | jq '.spec.taskSpec.steps[] | select(.name == "step-buildah-push" ) | .resources'
+{
+  "limits": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "requests": {
+    "cpu": "250m",
+    "memory": "65Mi"
+  }
+}
+```
+
+The pod definition is different, while Tekton will only use the **highest** values of one container, and set the rest(lowest) to zero:
+
+```sh
+$ kubectl -n test-build get pods buildah-golang-buildrun-9gmcx-pod-lhzbc -o json | jq '.spec.containers[] | select(.name == "step-step-buildah-bud" ) | .resources'
+{
+  "limits": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "requests": {
+    "cpu": "250m",
+    "ephemeral-storage": "0",
+    "memory": "65Mi"
+  }
+}
+
+$ kubectl -n test-build get pods buildah-golang-buildrun-9gmcx-pod-lhzbc -o json | jq '.spec.containers[] | select(.name == "step-step-buildah-push" ) | .resources'
+{
+  "limits": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "requests": {
+    "cpu": "0",               <------------------- See how the request is set to ZERO.
+    "ephemeral-storage": "0", <------------------- See how the request is set to ZERO.
+    "memory": "0"             <------------------- See how the request is set to ZERO.
+  }
+}
+```
+
+In this scenario, only one container can have the `spec.resources.requests` definition. Even when both steps have the same value, only one container will get them, the other will be set to zero.
```

Comment:
`Even when both steps have the same value` -> `Even when both steps have the same values` ?
`the other will be set to zero` -> `the others will be set to zero` ?

Reply:
fixed